Fixing includes, added dependency listing, and testing unit tests.

ollelogdahl opened the pull request:
Fixing includes: `splitwebp.cpp` requires `fstream` (`std::fstream` is defined in `fstream`). The `iostream` include has been replaced by `fstream`, since `fstream` itself includes `iostream`.

Linux also requires hdl5 installed separately (dependency of libopencv_hdl), and therefore i have added it in the dependency list in README.

Tried adding unit tests, but didn't have enough experience in cmake to make it work. I can develop unittests, if someone else sets up the environment! #3

ollelogdahl committed:
Adding gtests as ExternalProject in CMake (WIP)

ollelogdahl committed:
Changed to correct include for std::fstream. Listed hdf5 as dependency in README (required by libopencv_hdf).

ollelogdahl committed:
Added test_main, and removed all non working gtest in cmake.

Razdeep commented:
Inside the `tests` directory, we can have a separate `CMakeLists.txt` for tests, so that we could use `add_subdirectory()` from the root CMakeLists.txt . We can also omit the `test_main.cpp` by linking `gtest_main` to the target in the inner CMakeLists.

ollelogdahl committed:
Added Cmake for gtest (globally installed). Added simple unit tests.

ollelogdahl commented:
Now i have added a few unit tests. More code would need to be broken into multiple functions to be unit-tested easily. I Have also added a working Cmake gtest implementation (needing a global gtest install).

TODO: make gtest download & install locally.

Razdeep commented:
Here's a tip, whenever you are trying to contribute to any project,
don't make changes directly into the master branch.
You should create a separate branch every time.
Here's a quick tutorial [here](https://www.atlassian.com/git/tutorials/using-branches#:~:text=A%20branch%20represents%20an%20independent%20line%20of%20development.&text=The%20git%20branch%20command%20lets,checkout%20and%20git%20merge%20commands.) about git branches.

Razdeep commented on test/test_all.cpp:
```diff
@@ -0,0 +1,52 @@
+
+TEST(framesexporter_png, exportFrames_init) {
+    splitwebp::FrameExporter_PNG fe = splitwebp::FrameExporter_PNG();
+    ASSERT_EQ(fe.getOutputFileExtension, "png");
```
@ollelogdahl I believe you wanted to write `fe.getOutputFileExtension()` instead `fe.getOutputFileExtension` here.

Razdeep commented:
@ollelogdahl I think writing all tests in one file "test_all.cpp" would become a little troublesome once the project size grows.
I have seen that you have written test for `Util` and `SplitWebP`. May be we could split the large file "test_all.cpp" to smaller files inside `tests` directory like `Util_test`, `SplitWebP_test` and so on.

Razdeep commented on README.md:
```diff
@@ -18,6 +18,7 @@ A command line tool that helps in splitting webp animation into images
  - cmake
  - OpenCV 3
  - libwebp
+ - hdf5
```
If you observe the Linux Github Runner, it successfully builds there without an explicit installation of hdf5.
I think it automatically gets installed with OpenCV.
Maybe hdf5 was a missing dependency particular to your system.

ollelogdahl replied:
You're right. Technically, hdf5 is required by openCV, but for my part it didn't get installed automatically. Maybe it should be added as a note?

Razdeep replied:
Yes. We can write it as a note.
Did you install opencv using pacman?

ollelogdahl replied:
I did install it with pacman. OpenCV installed libopencv_hdf.so, but that one got linker errrors without manually installing hdf5. Weird!

ollelogdahl committed:
Split tests into multiple files. Added tests guide in README. Added the Cmake option for building tests.

Razdeep commented:
You Pull Request looks really nice.
I'm marking it as `hacktoberfest-accepted`.
Thank you for contributing. Shall I merge it now?

ollelogdahl commented:
Thank you! It was a fun project. will be keeping an eye open for other issues :)
Yes you can go ahead and merge!
